ehrql/tables/beta/tpp.py

```diff
@@ -42,23 +42,104 @@
 
 @table
 class addresses(EventFrame):
-    address_id = Series(int)
-    start_date = Series(datetime.date)
-    end_date = Series(datetime.date)
-    address_type = Series(int)
-    rural_urban_classification = Series(int)
-    imd_rounded = Series(int, constraints=[Constraint.ClosedRange(0, 32_800, 100)])
+    """
+    Addresses
+
+    Geographic characteristics of the home address a patient registers with a practice.
+    Each row in this table is one registration period per patient.
+    Occasionally, a patient has multiple active registrations on a given date.
+    The postcode from the address is mapped to an Output Area,
+    from which other larger geographic representations can be derived
+    (see various [ONS publications][addresses_ukgeographies] for more detail).
+
+    !!! tip
```

Comment:
This is a `!!! warning` in the current cohort-extractor docs but I thought it's more of a `!!! tip`? This code is also in our [How-to guides - Calculating each patient's IMD quintile](https://docs.opensafely.org/ehrql/how-to/examples/#finding-each-patients-imd-rank) but I thought it'd be good to highlight here too?

Reply:
We don't need this warning in ehrQL!  We can write zeros just fine, as shown by this test:

```
def test_zeros(tmp_path):
    filename = tmp_path / "file.arrow"
    column_specs = {
        "patient_id": ColumnSpec(int),
        "imd": ColumnSpec(int, min_value=0, max_value=32_800),
    }
    results = [
        (123, 0),
        (456, None),
        (789, 100),
    ]
    write_dataset(filename, results, column_specs)
    table = pyarrow.feather.read_table(filename)
    output_rows = [tuple(d.values()) for d in table.to_pylist()]
    assert output_rows == results
```

Reply:
oh thanks, I missed this! I deleted the warning but have kept the IMD tip here